### ISSUE TITLE
Plugin Template: mutex not initialized correctly

### ISSUE THREAD

**Issue by dsc28** [blackberry/WebWorks-Community-APIs#407]
I have written a Webworks plugin based off your excellent plugin template, in the course of that I found a bug: `template_ndk.hpp` defines two private member variables `cond` and `mutex` (see [here](https://github.com/blackberry/WebWorks-Community-APIs/blob/master/BB10-Cordova/Template/plugin/src/blackberry10/native/src/template_ndk.hpp#L58#L59)).

In the corresponding constructor in `template_ndk.cpp`, these two are supposed to get initialized, but they aren't. Instead, new variables, which are local to the constructor, are declared (see [here](https://github.com/blackberry/WebWorks-Community-APIs/blob/master/BB10-Cordova/Template/plugin/src/blackberry10/native/src/template_ndk.cpp#L33#L34)), which for me meant that the mutex was never correctly initialized, my thread synchronization acted weirdly and my app froze.

Long story short: in `template_ndk.cpp`, this:
```C++
		pthread_cond_t cond  = PTHREAD_COND_INITIALIZER;
		pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
```
should become this:
```C++
		cond  = PTHREAD_COND_INITIALIZER;
		mutex = PTHREAD_MUTEX_INITIALIZER;
```

Thanks for this (otherwise ;-) ) great template.

**Comment by timwindsor** [comment]
Yes I see that. We seem to have missed that in our tweaks to the template. Thanks for pointing it out.
